integration_tests/test_new_eo_staclike_metadata_fields.py

```diff
@@ -0,0 +1,241 @@
+    platform='landsat-5',
+    instrument='TM',
+    lat=None,
+    lon=None,
+    time=Range(datetime.strptime('2009-12-17 23:53:03.946464', "%Y-%m-%d %H:%M:%S.%f"),
+               datetime.strptime('2009-12-17 23:53:31.463325', "%Y-%m-%d %H:%M:%S.%f")),
+)
+
+
+def mk_measurement(m):
+    common = dict(dtype='int16',
+                  nodata=-999,
+                  units='1')
+
+    if isinstance(m, str):
+        return dict(name=m, **common)
+    elif isinstance(m, tuple):
+        name, dtype, nodata = m
+        m = common.copy()
+        m.update(name=name, dtype=dtype, nodata=nodata)
+        return m
+    elif isinstance(m, dict):
+        m_merged = common.copy()
+        m_merged.update(m)
+        return m_merged
+    else:
+        raise ValueError('Only support str|dict|(name, dtype, nodata)')
+
+
+def _create_sample_product_def(metadata_definition,
+                               measurements=('nbar_band01', 'nbar_band02', 'nbar_band03', 'nbar_band04', 'nbar_band05',
+                                             'nbar_band07', 'nbart_band01', 'nbart_band02', 'nbart_band03',
+                                             'nbart_band04', 'nbart_band05', 'nbart_band07', 'oa_azimuthal_exiting',
+                                             'oa_azimuthal_incident', 'oa_combined_terrain_shadow', 'oa_exiting_angle',
+                                             'oa_fmask', 'oa_incident_angle', 'oa_nbar_contiguity',
+                                             'oa_nbart_contiguity', 'oa_relative_azimuth', 'oa_relative_slope',
+                                             'oa_satellite_azimuth', 'oa_satellite_view', 'oa_solar_azimuth',
+                                             'oa_solar_zenith', 'oa_time_delta'),
+                               with_grid_spec=False,
+                               storage=None):
+
+    if storage is None and with_grid_spec is True:
+        storage = {'crs': 'epsg:4326'}
+
+    measurements = [mk_measurement(m) for m in measurements]
+
+    definition = dict(
+        name=metadata_definition['name'],
+        description=metadata_definition['description'],
+        metadata_type='eo',
+        metadata={},
+        measurements=measurements
+    )
+
+    if storage is not None:
+        definition['storage'] = storage
+
+    return definition
+
+
+def _get_product_info(metadata_definition):
+    fields = toolz.get_in(['dataset'], metadata_definition, {})
+    return {name: parse_dataset_field(value, name=name) for name, value in fields.items()
+            if name not in ('search_fields', 'grid_spatial')}
+
+
+def _convert_datetime(val):
+    try:
+        return val.strftime("%Y-%m-%d %H:%M:%S.%f")
+    except AttributeError:
+        return str(val)
+
+
+def _get_boundingbox(metadata_fname):
+    with open(str(metadata_fname)) as fl:
+        metadata = yaml.load(fl)
+    crs = CRS(metadata['crs'])
+    geo = Geometry(metadata['geometry'], crs=crs)
+    spatial_reference = str(crs.wkt)
```

Comment:
Use epsg code when available rather than wkt.

Eg.
```suggestion
    spatial_reference = 'epsg:{}'.format(crs.epsg) if crs.epsg else crs.wkt
```